src/components/DataMap/DataMap.js

```diff
@@ -181,26 +285,36 @@ class DataMap extends React.Component {
           {...this.props.isoline}  // update when any isoline prop changes
         />
 
-        <FeatureGroup>
+        <LayerControlledFeatureGroup
+          layers={this.state.geometryLayers}
+        >
           <EditControl
             position='topleft'
             draw={{
               marker: false,
               circle: false,
               polyline: false,
             }}
-            onCreated={this.handleAreaCreatedOrEdited}
-            onEdited={this.handleAreaCreatedOrEdited}
+            onCreated={this.handleAreaCreated}
+            onEdited={this.handleAreaEdited}
             onDeleted={this.handleAreaDeleted}
           />
-        </FeatureGroup>
+        </LayerControlledFeatureGroup>
+
+        <StaticControl position='topleft'>
+          <GeoLoader
+            onLoadArea={this.handleUploadArea}
+            title='Import polygon'
+          />
+        </StaticControl>
 
-        {
-          this.props.area &&
-          <GeoJSON
-            data={this.props.area}
+        <StaticControl position='topleft'>
+          {/* See comments above regarding current GeoExporter arrangement. */}
```

Comment:
Which comments is this referring to? I was hoping for an explanation of the empty {}, but did not find one.

Reply:
The only way to insert a comment inside a JSX component declaration is to start a code section (`{}`) that contains the comment(s). It's awkward, but it's better than no comments at all. So this comment refers to the next line